contrib/offline/manage-offline-files.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+CURRENT_DIR=$(cd $(dirname $0); pwd)
+OFFLINE_FILES_DIR="${CURRENT_DIR}/offline-files"
+FILES_LIST="${CURRENT_DIR}/temp/files.list"
+PATHS_LIST="${CURRENT_DIR}/temp/paths.list"
+NGINX_PORT=81
+
+# step 1 : generate files.list and paths.list by generate_list.sh
+./generate_list.sh
+
+# step 2 : download files
+rm -rf ${OFFLINE_FILES_DIR}
+mkdir  ${OFFLINE_FILES_DIR}
+wget -P ${OFFLINE_FILES_DIR} -i ${FILES_LIST}
+
+# step 3:  move files to the specified path of offline files
+while read -r line; do
+    FILE_NAME="$(echo ${line} | awk -F"/" '{print $NF}')"
+    FILE_PATH="${OFFLINE_FILES_DIR}/$(echo ${line} | sed s@"${FILE_NAME}"@@)"
+    mkdir -p ${FILE_PATH}
+    mv "${OFFLINE_FILES_DIR}/${FILE_NAME}" ${FILE_PATH}
+done <<< "$(cat ${PATHS_LIST})"
+
+# step 4 : run nginx container server
+sudo docker container inspect nginx >/dev/null 2>&1
+if [ $? -ne 0 ]; then
+    sudo docker run \
```

Comment:
On Kubernetes ecosystem, we are switching from `docker` to `nerdctl` to drop `docker` support.
This can be out of scope from this pull request because the existing scripts are still using `docker` command.
just wanted to confirm what should be our direction.

Reply:
@oomichi  Ok, I see, I can create a separate PR later if we need to switch, since (AFAIK) `manage-offline-container-images.sh` also uses docker and I can modify it together. 🤔

Reply:
That makes sense, skopeo looks like a good option, I'm interested in it and plan to look into it

Reply:
Nice information about skopeo.
I checked it a little bit, that seems only for managing container images.
We cannot replace `docker run`/`nerdctl run` here with skopeo, right?

Reply:
my understanding of the comments above is that there are two todos:
1. switch from `docker run` to `nerdctl run`
2. use skopeo for images synchronization to replace the current `pull / tag / push` operations of images in `manage-offline-container-images.sh`

Reply:
I'll create new PRs to deal with these todos later

Reply:
> Nice information about skopeo. I checked it a little bit, that seems only for managing container images. We cannot replace `docker run`/`nerdctl run` here with skopeo, right?

Indeed not, skopeo is only for managing oci images, to run them you need podman.

Reply:
@floryut which is better for our project, nerdctl or podman?

Reply:
I would say podman, as stated before nerdctl is bit containerd oriented and probably not available/wanted by crio users.
podman/skopeo/crun/buildah are all more OCI oriented 😄

Reply:
I have switched `docker run` to `podman run` in manage-offline-files.sh 😊
PTAL @floryut @oomichi @cristicalin

Reply:
we can use code like "https://github.com/automotiveMastermind/numonic/blob/main/hack/build.sh#L91" to support podman , nerdctl and even docker together.

Reply:
PTAL @floryut @oomichi @cristicalin 🤠

Reply:
Wow this became a long thread. The reason we have the `docker run` and `nerdctl run` command in there to fetch files is because we don't have an elegant way to get files out of container images so what we do is `$command run <image> -v /tmp:/tmp cp <whatever> /tmp/<whatever>` and get the file from the container, it is rather inelegant. My understanding was that nerdctl recently (0.19.0 I think) got the capability to copy files directly from images without the need to run them, maybe skopeo has the same since it is an image manipulation tool.

Overall out current download, cache and offline logic could use a full overhaul with a clean multi-step separation and runtime agnostic code. By this I mean always check the cache for needed items, if items are not in the cache fetch them in a runtime agnostic way and provide a way to populate the cache so that no external connection is needed if the cache is already complete.

Reply:
@oomichi 😄 are there any additional suggestions?